tools/oeedger8r/CMakeLists.txt

```diff
@@ -6,22 +6,89 @@
 # do not emit to the current working directory, they always emit to
 # the location of the input file.
 
-find_program(ESY esy)
-if (NOT ESY)
-  message(FATAL_ERROR "Please check your esy installation")
-endif ()
+if (USE_PLATFORM_OCAML)
+  # Use ocaml compilers if available.
+  # Use ocaml compilers if available
+  find_program(OCAMLLEX ocamllex)
+  find_program(OCAMLYACC ocamlyacc)
+  find_program(OCAMLOPT ocamlopt)
+  if ((NOT OCAMLLEX) OR (NOT OCAMLYACC) OR (NOT OCAMLOPT))
+     message(FATAL_ERROR "ocamllex, ocamlyacc or ocamlopt not found.\n"
+                         "Install ocaml via your platform's package manager.")
+  endif ()
+
+  # Generate Lexer.
+  add_custom_command(
+    OUTPUT Lexer.ml
+    COMMAND ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_SOURCE_DIR}/intel/Lexer.mll ${CMAKE_CURRENT_BINARY_DIR}
+    COMMAND ocamllex Lexer.mll
+    DEPENDS intel/Lexer.mll)
+
+  # Generate Parser.
+  add_custom_command(
+    OUTPUT Parser.ml Parser.mli
+    COMMAND ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_SOURCE_DIR}/intel/Parser.mly ${CMAKE_CURRENT_BINARY_DIR}
+    COMMAND ocamlyacc Parser.mly
+    DEPENDS intel/Parser.mly)
+
+  # Compile
```

Comment:
Consider using a function/macro for these similar ocamlopt commands (similar to the Makefile pattern).